components/camel-properties/src/main/java/org/apache/camel/component/properties/DefaultPropertiesParser.java

```diff
@@ -318,7 +318,14 @@ private String doGetPropertyValue(String key) {
                 }
             }
             if (value == null && envMode == PropertiesComponent.ENVIRONMENT_VARIABLES_MODE_OVERRIDE) {
-                value = System.getenv(key);
+                // lookup OS env with upper case key
```

Comment:
@davsclaus , what do you think about to move common code to the separate function?

Reply:
Yeah we can do that